### ISSUE TITLE
Agent 2 value function not learned?

### ISSUE THREAD

**Issue by cooijmanstim** [Silent-Zebra/POLA#8]
Here's another bug I found. `out_lookahead` receives `trainstate_val2_copy.params` rather than the argument `trainstate_val2_copy_params` that the containing function is being differentiated with respect to. As a consequence I think agent 2's value function is not being trained at all.

The analogous code for agent 1 (line 961) does not have this issue.

https://github.com/Silent-Zebra/POLA/blob/6b07e89317b07d91216db9d02c1f915f9313b66a/jax_files/POLA_dice_jax.py#L1061

**Comment by Silent-Zebra** [comment]
Thanks for pointing this out! I really need to use more different variable names that are less likely to lead to such mistakes in the future. I guess the coin game is nice because the rewards tend to be close-ish to 0 (especially at the start with close to random policies) so the value function doesn't matter that much. This probably explains why the method still mostly works. Fixing this should improve performance specifically for agent 2.

**Comment by cooijmanstim** [comment]
It might matter more than you think. If the value function were used as just a REINFORCE baseline then I would agree that it won't matter much. However it is used in the advantage as a proxy for the return, so agent 2 is optimizing a nonsensical objective.

**Comment by Silent-Zebra** [comment]
Closing as fixes have been implemented and experiments rerun (see updated readme)
